core/config/project_settings.cpp

```diff
@@ -56,6 +58,79 @@ String ProjectSettings::get_resource_path() const {
 
 const String ProjectSettings::IMPORTED_FILES_PATH("res://.godot/imported");
 
+// Returns the features that a project must have when opened with this build of Godot.
+// This is used by the project manager to provide the initial_settings for config/features.
+PackedStringArray ProjectSettings::get_required_features() {
+	PackedStringArray features = PackedStringArray();
+	features.append(VERSION_BRANCH);
+#ifdef REAL_T_IS_DOUBLE
+	features.append("Double");
+#else
+	features.append("Single");
```

Comment:
If there's enough space, I think it'd be better to be explicit that this is about precision:

```suggestion
	features.append("DoublePrecision");
#else
	features.append("SinglePrecision");
```

Alternatively, this could be `DoublePhysics` and `SinglePhysics` respectively, but this could be confused for a physics engine.

Reply:
Changed to end in `-precision`. Testing this also made me discover a bug caused by #45968 as now I have to explicitly call `save();` when updating the project features, so that's fixed now :)